### PR TITLE
ADD stock_quant_sale_order

### PR DESCRIPTION
**Pull request by mra-odoo**
- Select some stock quants from the list view
- Got to Action > Create Sales Order and the selected quants will be validated 
	i) All selected stock quants should be in the same location
- A wizard shows up and allow user to input the customer.
- Upon pressing "Create Sales Order", a draft sales order should be created with the selected stock quants as the order lines.
** Please consider company_id of the user when selecting the records for creating the sales order.

### CODE REVIEW

File: stock_quant_sale_order/wizard/stock_quant_sale_order.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        stock_quant_obj = self.env['stock.quant']
+        sale_order_obj = self.env['sale.order']
+        sale_order_line_obj = self.env['sale.order.line']
+
+        active_ids = context.get('active_ids', [])
+        quant_ids = stock_quant_obj.browse(active_ids)
+        source_location = quant_ids[0].location_id
+
+        if any(q.location_id != source_location for q in quant_ids):
+            raise UserError(_('Please select quants that are in the same '
```
Comment (TimLai125) [inline stock_quant_sale_order/wizard/stock_quant_sale_order.py]: Can we move this check before the wizard view is opened?

File: stock_quant_sale_order/wizard/stock_quant_sale_order.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                              'Location.'))
+
+        if any(q.sale_order_id != False for q in quant_ids):
```
Comment (TimLai125) [inline stock_quant_sale_order/wizard/stock_quant_sale_order.py]: This condition seems to be always "True", can you kindly check the behavior?

File: stock_quant_sale_order/wizard/stock_quant_sale_order.py (hunk 1 above)
Comment (mra-odoo) [reply stock_quant_sale_order/wizard/stock_quant_sale_order.py]: Fixed.

File: stock_quant_sale_order/wizard/stock_quant_sale_order.py (hunk 2 above)
Comment (mra-odoo) [reply stock_quant_sale_order/wizard/stock_quant_sale_order.py]: Fixed.

**Comment by mra-odoo** [comment]
We have also fixed :

 _** Please consider company_id of the user when selecting the records for creating the sales order.**_

**Comment by mra-odoo** [comment]
Fixed now so before opening wizard it will raise warning.

**Pull request closed without merging**